test/with_dummyserver/test_socketlevel.py

```diff
@@ -843,6 +843,77 @@ def socket_handler(listener):
         pool = HTTPConnectionPool(self.host, self.port, retries=False)
         pool.request('GET', '/', headers=HTTPHeaderDict(headers))
         self.assertEqual(expected_headers, parsed_headers)
+    
+    def test_request_headers_are_sent_in_the_original_order(self):
+        # NOTE: Probability this test gives a false negative is 1/(K!)
+        K = 16
+        # NOTE: Provide headers in non-sorted order (i.e. reversed)
+        #       so that if the internal implementation tries to sort them,
+        #       a change will be detected.
+        expected_request_headers = [(u'X-Header-%d' % i, str(i)) for i in reversed(range(K))]
+        
+        actual_request_headers = []
+
+        def socket_handler(listener):
+            sock = listener.accept()[0]
+
+            buf = b''
+            while not buf.endswith(b'\r\n\r\n'):
+                buf += sock.recv(65536)
+
+            headers_list = [header for header in buf.split(b'\r\n')[1:] if header]
+
+            for header in headers_list:
+                (key, value) = header.split(b': ')
+                if not key.decode('ascii').startswith(u'X-Header-'):
+                    continue
+                actual_request_headers.append((key.decode('ascii'), value.decode('ascii')))
+
+            # Send incomplete message (note Content-Length)
```

Comment:
This comment is a little bit confusing. I'm aware it's wrong in a few other places, but you don't need to fix those up: let's just remove it from here.